src/test/java/com/commercetools/sync/services/impl/UnresolvedReferencesServiceImplTest.java

```diff
@@ -0,0 +1,197 @@
+package com.commercetools.sync.services.impl;
+
+import com.commercetools.sync.commons.models.WaitingToBeResolved;
+import com.commercetools.sync.products.ProductSyncOptions;
+import com.commercetools.sync.products.ProductSyncOptionsBuilder;
+import io.sphere.sdk.client.BadRequestException;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.CustomObjectDraft;
+import io.sphere.sdk.customobjects.commands.CustomObjectUpsertCommand;
+import io.sphere.sdk.products.ProductDraft;
+import io.sphere.sdk.queries.PagedQueryResult;
+import io.sphere.sdk.utils.CompletableFutureUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class UnresolvedReferencesServiceImplTest {
+
+    private UnresolvedReferencesServiceImpl service;
+    private ProductSyncOptions productSyncOptions;
+    private List<String> errorMessages;
+    private List<Throwable> errorExceptions;
+
+    @BeforeEach
+    void setUp() {
+        errorMessages = new ArrayList<>();
+        errorExceptions = new ArrayList<>();
+        productSyncOptions = ProductSyncOptionsBuilder.of(mock(SphereClient.class))
+                .errorCallback((errorMessage, errorException) -> {
+                    errorMessages.add(errorMessage);
+                    errorExceptions.add(errorException);
+                })
+                .build();
+        service = new UnresolvedReferencesServiceImpl(productSyncOptions);
+    }
+
+    @Test
+    void save_WithSuccessfulMockCtpResponse_ShouldReturnMock() {
+        final CustomObject mock = mock(CustomObject.class);
```

Comment:
Here the mock var is called `mock`, but later you have `final ProductDraft productDraft = mock(ProductDraft.class);`. Pls make it consistent.

Reply:
95578f1b9bfe5a0363c6be232839f7fd23bda287